test/connectivity/README.md

```diff
@@ -69,3 +72,106 @@ Network offline/online should not affect this test result since Edge hub stores
+
+#### Cloud to Module Direct Method Test
+
+![Connectivity Test Diagram](./images/ConnectivityTest_C2M_DirectMethod.png)
+
+The Direct Method Connectivity Test is designed to test direct method resilience against network connectivity issue. 
+##### Test Specific Component:
+	1. Direct Method (Cloud) Sender is an Azure service client that is wrapped in a container module that is deployed into a test agent.  It is responsible for sending a "HelloWorldMethod" direct method to the Direct Method Receiver module and report the direct method status to Test Result Coordinator
+	2. Direct Method Receiver is a module container that runs a module client that is deployed into a test agent. It is a targeted receiver of Direct Method Sender. It is responsible for replying to the "HelloWorldMethod" direct method and notify Test Result Coordinate if it receives a direct method.
+
+##### Test Mechanism: 
+	1. The Direct Method Cloud Sender invokes a test direct method to the IoT Hub
+	2. The IoT Hub forwards the direct method to Edge Hub
+	3. The Edge Hub forwards the direct method to Direct Method Receiver
+	4. The Direct Method Receiver notifies Test Analyzer regarding the received direct method
+	5. The Direct Method Receiver responds to Edge Hub's direct method invocation
+	6. The Edge Hub  responds to IoT Hub direct method invocation 
+	7. The Direct Method Cloud Sender receives a response from IoT Hub
+	8. The Direct Method Cloud Sender notifies Test Analyzer regarding its direct method response
+Test Analyzer compile a report from Direct Method Cloud Sender, Direct Method Receiver, Network Controller and determine if the test case is passed or failed.
+
+#### Cloud to EdgeAgent Direct Method Test (PING)
+
+![Connectivity Test Diagram](./images/ConnectivityTest_Ping_DirectMethod.png)
+
+##### Test Specific Component:
+	1. Direct Method (Cloud) Sender is an Azure service client that is wrapped in a container module that is deployed into a test agent.  It is responsible for sending a "HelloWorldMethod" direct method to the EdgeAgent module and report the direct method status to Test Result Coordinator
+
+##### Test Mechanism: 
+	1. The Direct Method Cloud Sender invokes a test direct method to the IoT Hub
+	2. The IoT Hub forwards the direct method to Edge Agent
+	3. The Edge Agent has a pre-implemented "ping" method with a response.
+	4. The Direct Method Cloud Sender receives a response from IoT Hub
+	5. The Direct Method Cloud Sender notifies Test Analyzer regarding its direct method response
+Test Analyzer compile a report from Direct Method Cloud Sender, Direct Method Receiver, Network Controller and determine if the test case is passed or failed. 
+
+#### Cloud to Device Message Test
+
+![Connectivity Test Diagram](./images/ConnectivityTest_C2D_Message.png)
+
+The goal of the test scheme is to verify the recoverability of the cloud to device messaging once the device goes offline. 
```

Comment:
should we add a section called "Test goal" for all tests?

Reply:
I don't know. I'll leave it up to your call :/

Reply:
leave it for now.